<launch>

	<arg name="deploy_dir" default="/home/jialiang/Code/thesis_ws/src/adaptive_open_local_planner"/>
    
    <node name="move_base" pkg="adaptive_open_local_planner" type="test_planner" output="screen" args="--log_dir $(arg deploy_dir)/log --minloglevel 0 --max_log_size 100 --alsologtostderr" launch-prefix="xterm -e gdb -ex run --args ">
        <rosparam file="$(find adaptive_open_local_planner)/param/costmap_common_params.yaml" command="load" ns="global_costmap" /> 
        <rosparam file="$(find adaptive_open_local_planner)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find adaptive_open_local_planner)/param/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find adaptive_open_local_planner)/param/global_costmap_params.yaml" command="load" /> 
        <rosparam file="$(find adaptive_open_local_planner)/param/base_local_planner_params.yaml" command="load" />
        <rosparam file="$(find adaptive_open_local_planner)/param/move_base_params.yaml" command="load" />
        <rosparam file="$(find adaptive_open_local_planner)/param/parameters.yaml" command="load" />
        <rosparam file="$(find adaptive_open_local_planner)/param/plugins.yaml" command="load" ns="/costmap_node/costmap"/>
        <!-- <param name="base_global_planner" value="HybridAStar/HybridAStarPlanner" />这个标志是选择使用特定的全局规划器 -->
        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
        <!-- <param name="use_adaptive_open_local_planner" value="true" /> -->
    </node>
    <node name="tf_broadcaster" pkg="adaptive_open_local_planner" type="tf_test_broadcaster" output="screen" />
    <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(find hybrid_astar)/maps/mymap2077.yaml" /> -->
        <node name="map_server" pkg="map_server" type="map_server" args="$(find hybrid_astar)/maps/parking_lot_8_5_36_resize.yaml" />
        <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(find hybrid_astar)/maps/map.yaml" /> -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find hybrid_astar)/rviz/config_movebase.rviz" />
</launch>